prody/dynamics/plotting.py

```diff
@@ -2381,3 +2355,39 @@ def showTree_networkx(tree, node_size=20, node_color='red', node_shape='o',
         showFigure()
 
     return mpl.gca()
+
+
+def checkColors(colors, num, labels, allowNumbers=False):
+    """Check colors and process them if needed"""
+
+    from matplotlib.colors import is_color_like
+
+    colors_dict = {}
+
+    if is_color_like(colors) or colors is None or isinstance(colors, Number):
+        colors = [colors] * num
+    elif isListLike(colors):
+        colors = list(colors)
+    elif isinstance(colors, dict):
+        if labels is None:
+            raise TypeError('color cannot be a dict unless labels are provided')
+        colors_dict = colors
+        colors = [colors_dict[label] for label in labels]
+
+    if isinstance(colors, list):
+        if len(colors) != num:
+            raise ValueError('colors should have the length of the set to be colored or satisfy matplotlib color rules')
+
+        for color in colors:
+            if not is_color_like(color):
+                if not allowNumbers:
```

Comment:
I think if `allowNumbers`, you may need to convert the number to the color in cycle like this:
```
cycle_colors = plt.rcParams['axes.prop_cycle'].by_key()['color']
color = cycle_colors[color % len(cycle_colors)]
```

Reply:
So the entire loop may look like this:
```
if isinstance(colors, list):
        if len(colors) != num:
            raise ValueError('colors should have the length of the set to be colored or satisfy matplotlib color rules')

        for i, color in enumerate(colors):
            if not is_color_like(color):
                if allowNumbers:
                    cycle_colors = plt.rcParams['axes.prop_cycle'].by_key()['color']
                    colors[i] = cycle_colors[color % len(cycle_colors)]
                else:
                    raise ValueError('each element of colors should satisfy matplotlib color rules')
       ....
```

Reply:
I think there's a conversion of the numbers with the color cycle somewhere else, but yes, we could maybe move it here.

Reply:
Actually, we use matplotlib.colors.Normalize instead to link up with the cmap. This is on lines 356 and 428

This could somehow be incorporated too though